elementary/monitor/data_monitoring/report/data_monitoring_report.py

```diff
@@ -221,22 +250,69 @@ def send_report(
             if not send_succeded:
                 self.success = False
 
-        if self.s3_client:
-            send_succeded = self.s3_client.send_report(
+        self.execution_properties["success"] = self.success
+        return self.success
+
+    def upload_report(
+        self, local_html_path: str, remote_file_path: Optional[str] = None
+    ) -> Tuple[bool, Optional[str]]:
+        if self.gcs_client:
+            send_succeded, bucket_website_url = self.gcs_client.send_report(
                 local_html_path, remote_bucket_file_path=remote_file_path
             )
-            self.execution_properties["sent_to_s3_successfully"] = send_succeded
+            self.execution_properties["sent_to_gcs_successfully"] = send_succeded
             if not send_succeded:
                 self.success = False
 
-        if self.gcs_client:
-            send_succeded = self.gcs_client.send_report(
+        if self.s3_client:
+            send_succeded, bucket_website_url = self.s3_client.send_report(
                 local_html_path, remote_bucket_file_path=remote_file_path
             )
-            self.execution_properties["sent_to_gcs_successfully"] = send_succeded
+            self.execution_properties["sent_to_s3_successfully"] = send_succeded
             if not send_succeded:
                 self.success = False
 
+        self.execution_properties["success"] = self.success
+        return self.success, bucket_website_url
+
+    def send_test_results_summary(
+        self,
+        days_back: Optional[int] = None,
+        test_runs_amount: Optional[int] = None,
+        disable_passed_test_metrics: bool = False,
+        bucket_website_url: Optional[str] = None,
+        include_description: bool = False,
+    ) -> bool:
+        if self.slack_client:
```

Comment:
It's a bit confusing to check if `self.slack_client` is defined here, as the code does nothing if it's not.

I think it's clearer to do it in the calling code instead (and fail here if this function is called without a slack client being set).